### PR TITLE
feat(query): Add request options to search/read methods

### PR DESCRIPTION
**Pull request by ElPicador**

### CODE REVIEW

File: algoliasearch-common/src/main/java/com/algolia/search/APIClient.java [hunk 1]
```diff
@@ -54,6 +55,20 @@
   }
 
   /**
+   * List all existing indexes
+   *
+   * @return A List of the indices and their metadata
+   * @throws AlgoliaException
+   */
+  public List<Index.Attributes> listIndices(RequestOptions requestOptions) throws AlgoliaException {
```
Comment (LeoErcolanelli) [inline algoliasearch-common/src/main/java/com/algolia/search/APIClient.java]: I may be missing something, but requestOptions is not used, right?

Comment (ElPicador) [reply algoliasearch-common/src/main/java/com/algolia/search/APIClient.java]: You are right, this PR was only the start of a POC.
So I didn't implement everything yet

Comment (LeoErcolanelli) [reply algoliasearch-common/src/main/java/com/algolia/search/APIClient.java]: My bad, I missed the "early" before "implementation" in Clément's mail

**Review by clement-leprovost — APPROVED** [review]
LGTM 👍 

It's interesting that you chose to go for a `@NonNull` argument, and specify `RequestOptions.empty`. Is there a specific reason? It's true that explicitly passing "empty" makes the code a tad more readable. :)

**Comment by ElPicador** [comment]
The second thing I hate the most are `null` and `NullPointerException` so I try to enforce no `null` and use different patterns (empty objects, `Option`, etc.).

**Pull request merged**